### PR TITLE
e2e tests

### PR DESCRIPTION
**Pull request by brybacki**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
First push - work in progress, more tests will follow

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: e2e/create-pvc_test.go [hunk 1]
```diff
@@ -67,69 +70,71 @@ var _ = Describe("CreatePVC", func() {
...
 		Expect(err).ToNot(HaveOccurred())
 
-		By("creating a pod that attaches pvc")
+		pod := podWithPvcSpec(podName, pvcName, cmd, args)
+		// add toleration so we can use control node for tests
+		pod.Spec.Tolerations = []k8sv1.Toleration{{
+			Key:      "node-role.kubernetes.io/master",
+			Operator: k8sv1.TolerationOpExists,
+			Effect:   k8sv1.TaintEffectNoSchedule,
+		}}
```
Comment (davidvossel) [inline e2e/create-pvc_test.go:134]: it's probably fine to simply add this to all pods created by the generic podWithPvcSpec function

File: e2e/create-pvc_test.go [hunk 2]
```diff
@@ -67,69 +70,71 @@ var _ = Describe("CreatePVC", func() {
...
+		Eventually(func() error {
+			updatedPod, err := tenantClient.CoreV1().Pods(namespace).Get(context.Background(), newPod.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if updatedPod.Status.Phase != k8sv1.PodRunning {
+				return fmt.Errorf("Pod in phase %s, expected Running", updatedPod.Status.Phase)
+			}
+			return nil
+		}, 3*time.Minute, 5*time.Second).Should(Succeed(), "pod should reach running state")
```
Comment (davidvossel) [inline e2e/create-pvc_test.go:101]: we could probably have helper functions for waiting for the pod to hit a running state, and another one for waiting for a pod to be deleted.

File: e2e/create-pvc_test.go [hunk 3]
```diff
@@ -67,69 +70,71 @@ var _ = Describe("CreatePVC", func() {
...
+
+	})
+	//       kubernetes.io/hostname: kvcluster-control-plane-z6jcv
+
+	It("creates a pvc and attaches to pod", Label("pvcCreation"), func() {
```
Comment (davidvossel) [inline e2e/create-pvc_test.go:106]: this test needs a unique description

File: e2e/create-pvc_test.go (hunk 2 above)
Comment (brybacki) [reply e2e/create-pvc_test.go:101]: yes

File: e2e/create-pvc_test.go (hunk 1 above)
Comment (brybacki) [reply e2e/create-pvc_test.go:134]: ok

File: e2e/create-pvc_test.go (hunk 3 above)
Comment (brybacki) [reply e2e/create-pvc_test.go:106]: right

**Review by davidvossel — COMMENTED** [review]
/lgtm
/approve

thanks!

**Pull request merged**